src/pybind/mgr/osd_support/module.py

```diff
@@ -0,0 +1,284 @@
+            self.check_osds.update(self.osd_ids)
+
+            report = list()
+            for osd_id in self.check_osds:
+                pgs = self.get_pg_count(osd_id)
+                report.append(dict(osd_id=osd_id, pgs=pgs))
+            out = f"{report}"
+
+        elif cmd_prefix == 'osd drain stop':
+            if not osd_ids:
+                self.log.debug("No osd_ids provided, stop all pending drain operations)")
+                self.osd_ids = set()
+                self.emptying_osds = set()
+
+                # this is just a poor-man's solution as it will not really stop draining
+                # the osds. It will just stop the queries and also prevent any scheduled OSDs
+                # from getting drained at a later point in time.
+                out = "Stopped all future draining operations (not resetting the weight for already reweighted OSDs)"
+
+            else:
+                if not_found_osds:
+                    return -errno.EINVAL, '', f"OSDs <{not_found_osds}> not found in cluster"
+
+                self.osd_ids = self.osd_ids.difference(osd_ids)
+                self.emptying_osds = self.emptying_osds.difference(osd_ids)
+                out = f"Stopped draining operations for OSD(s): {osd_ids}"
+
+        else:
+            return HandleCommandResult(
+                retval=-errno.EINVAL,
+                stdout='',
+                stderr=f"Command not found <{cmd.get('prefix', '')}>")
+        return HandleCommandResult(
+            retval=ret,   # exit code
+            stdout=out,   # stdout
+            stderr=err)
+
+    def serve(self):
+        """
+        This method is called by the mgr when the module starts and can be
+        used for any background activity.
+        """
+        self.log.info("Starting mgr/osd_support")
+        while self.run:
+            # Do some useful background work here.
+
+            self.log.debug(f"Scheduled for draining: <{self.osd_ids}>")
+            self.log.debug(f"Currently being drained: <{self.emptying_osds}>")
+            # the state should be saved to the mon store in the actual call and
+            # then retrieved in serve() probably
+
+            # 1) check if all provided osds can be stopped, if not, shrink list until ok-to-stop
+            for x in self.find_osd_stop_threshold(self.osd_ids):
+                self.emptying_osds.add(x)
+
+            # remove the emptying osds from the osd_ids since they don't need to be checked again.
+            self.osd_ids = self.osd_ids.difference(self.emptying_osds)
+
+            # 2) reweight the ok-to-stop osds, ONCE
+            self.reweight_osds(self.emptying_osds)
+
+            # 3) check for osds to be empty
+            empty_osds = self.empty_osds(self.emptying_osds)
+
+            # remove osds that are marked as empty
+            self.emptying_osds = self.emptying_osds.difference(empty_osds)
+
+            # fixed sleep interval of 10 seconds
+            sleep_interval = 10
```

Comment:
might be useful to have this as a configurable at some point

Reply:
Initially I copied the boilerplate code from the `hello` module, which implements configurable sleep intervals, which I dropped in an earlier version (see https://github.com/ceph/ceph/pull/32677#discussion_r369005195). Unless you have a real strong reason to add it back, I'd keep it that way.

Reply:
I am cool with it not being a configurable, as long as you think 10 seconds is all you'll ever need.